pkg/build/api/v1beta1/types.go

```diff
@@ -223,6 +228,17 @@ type WebHookTrigger struct {
 	Secret string `json:"secret,omitempty" yaml:"secret,omitempty"`
 }
 
+// BuildTriggerImageChangeParams represents the parameters to the ImageChange trigger.
+type ImageChangeTrigger struct {
+	// Image is used to specify the value in the BuildConfig to replace with the 
+	// immutable image id supplied by the ImageRepository when this trigger fires.
+	Image string `json:"image,omitempty" yaml:"image,omitempty"`
+	// ImageRepositoryRef a reference to a Docker image repository to watch for changes.
+	ImageRepositoryRef kapi.ObjectReference `json:"imageRepositoryRef,omitempty" yaml:"imageRepositoryRef,omitempty"`
+	// Tag is the name of an image repository tag to watch for changes.
```

Comment:
I haven't looked at all the other use cases, but in this use case, the only way we can make it non-optional is if we default to "latest".  i'm fine with doing that if you prefer it, but there's no way to perform image change trigger logic without some tag value to be watching for changes.